### PR TITLE
Add offer modal base

### PR DESCRIPTION
**Pull request by juhakujala**
localhost_3000_customers_1005

### CODE REVIEW

File: src/components/offer/OfferModal.module.scss [hunk 1]
```diff
@@ -0,0 +1,111 @@
+.offerDialog {
+  @media (max-width: 1247px) { // --breakpoint-l minus 1
+    width: 100% !important;
+  }
+
+  @media (min-width: 1248px) { // --breakpoint-l
```
[the remark this answers is not in the document]
Comment (juhakujala) [reply src/components/offer/OfferModal.module.scss:6]: That is true. This is a copy-paste error, so I expect some other files have the same issue 😅

File: src/components/offer/OfferModal.tsx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+            </>
+          ) : (
+            <th>{t(`${T_PATH}.familyWithChildren`)}</th>
+          )}
+        </tr>
+      </thead>
+      <tbody className="hds-table__content">
+        <tr>
+          <td>
+            <span className={styles.emphasized}>{apartment?.apartment_number}</span>
```
[the remark this answers is not in the document]
Comment (juhakujala) [reply src/components/offer/OfferModal.tsx:93]: I added line 44 check after I already had all the `project?` and others in place. I will remove those 👍

File: src/components/offer/OfferModal.module.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (juhakujala) [reply src/components/offer/OfferModal.module.scss]: Added a new commit to fix other files as well

**Comment by juhakujala** [comment]
> Tried this locally, and it looks fancy, but there seemed to be some problem related to filling the fields: every time I added something to a field, like one more character to a text field, the focus shifted from the field to the form itself, making filling the fields very annoying 😄 Happened with both Chrome and Firefox.

I know that this still has an issue. However, I did not use that much time at this point trying to get this solution to work 100% as we don't even yet know how we are going to handle this form's data in the near future. So I kinda left it like this for now until we actually get the initial form data from the backend and know what kind of data it is and how to send this form.

This focus issue has to do with React's `useState` which re-renders the component after the state changes. And now that we are storing the form values to the component state and HDS Dialog is made accessible, it focuses the dialog title on each re-render making it lose focus on the input fields. We could get past this by removing the toggles and trying to go without using the component state.

Nevertheless, this was just a dummy component to start with, so we can improve this when we hook it up to the backend.

**Review by tuomas777 — APPROVED** [review]
LGTM 👍 🐰

**Pull request merged**